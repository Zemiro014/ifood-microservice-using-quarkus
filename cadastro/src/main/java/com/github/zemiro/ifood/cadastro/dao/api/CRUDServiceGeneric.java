package com.github.zemiro.ifood.cadastro.dao.api;

import com.github.zemiro.ifood.cadastro.entities.Restaurente;

import java.util.List;

public abstract class  CRUDServiceGeneric<T, D> {
    public abstract List<T> findAll();
    public abstract void delete(Long id);
    public abstract void update(Long id, D dto);
    public abstract void create(T entity);
}