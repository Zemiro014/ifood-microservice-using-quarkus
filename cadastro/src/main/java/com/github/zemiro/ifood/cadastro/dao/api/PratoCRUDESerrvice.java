package com.github.zemiro.ifood.cadastro.dao.api;

import com.github.zemiro.ifood.cadastro.entities.Prato;
import com.github.zemiro.ifood.cadastro.entities.Restaurente;
import org.jboss.resteasy.annotations.jaxrs.PathParam;

import java.util.List;

public interface PratoCRUDESerrvice{
    public List<Restaurente> findDishByRestaurant(Long restauranteId);
    public void createNewDishToRestaurante(Long restauranteId, Prato dto);
    public void updateDishOfRestaurante(Long restauranteId, Long pratoId, Prato dto);
    public void deleteDishFromRestaurant( Long restauranteId, Long pratoId);
}