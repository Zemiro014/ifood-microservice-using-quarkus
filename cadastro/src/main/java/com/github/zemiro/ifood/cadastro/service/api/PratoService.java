package com.github.zemiro.ifood.cadastro.service.api;

import com.github.zemiro.ifood.cadastro.entities.Prato;
import com.github.zemiro.ifood.cadastro.entities.Restaurente;

import java.util.List;

public interface PratoService {
    public List<Restaurente> findDishByRestaurant(Long restauranteId);
    public void createNewDishToRestaurante(Long restauranteId, Prato dto);
    public void updateDishOfRestaurante(Long restauranteId, Long pratoId, Prato dto);
    public void deleteDishFromRestaurant(Long restauranteId, Long pratoId);
}