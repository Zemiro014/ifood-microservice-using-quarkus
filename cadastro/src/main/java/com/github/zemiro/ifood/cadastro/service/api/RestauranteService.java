package com.github.zemiro.ifood.cadastro.service.api;

import com.github.zemiro.ifood.cadastro.entities.Restaurente;

import java.util.List;

public interface RestauranteService {
    public List<Restaurente> findAllRestaurant();
    public void deleteRestaurant(Long id);
    public void updateRestaurant(Long id, Restaurente dto);
    public void createRestaurant(Restaurente entity);
}
