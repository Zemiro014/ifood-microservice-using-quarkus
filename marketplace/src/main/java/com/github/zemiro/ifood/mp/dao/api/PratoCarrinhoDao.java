package com.github.zemiro.ifood.mp.dao.api;

import com.github.zemiro.ifood.mp.referenceEntity.PratoCarrinho;
import io.smallrye.mutiny.Uni;
import io.vertx.mutiny.pgclient.PgPool;

import java.util.List;

public interface PratoCarrinhoDao {
    public Uni<String> savePratoCarrinho(PgPool pgPool, String cliente, Long prato);

    public Uni<List<PratoCarrinho>> findPratoCarrinho(PgPool pgPool, String cliente);

    public Uni<Boolean> deletePratoCarrinho(PgPool client, String cliente);
}
