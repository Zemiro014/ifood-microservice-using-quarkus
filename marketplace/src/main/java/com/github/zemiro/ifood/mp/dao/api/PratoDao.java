package com.github.zemiro.ifood.mp.dao.api;

import com.github.zemiro.ifood.mp.dto.PratoDTO;
import io.smallrye.mutiny.Multi;
import io.vertx.mutiny.pgclient.PgPool;

public interface PratoDao {
    public Multi<PratoDTO> findAll(PgPool pgPool);
    public Multi<PratoDTO> findAll(PgPool pgPool, Long idRestaurante);
    public Multi<PratoDTO> findById(PgPool pgPool, Long id);
}