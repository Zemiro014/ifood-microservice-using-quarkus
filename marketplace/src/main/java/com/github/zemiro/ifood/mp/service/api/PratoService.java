package com.github.zemiro.ifood.mp.service.api;

import com.github.zemiro.ifood.mp.dto.PratoDTO;
import io.smallrye.mutiny.Multi;
import io.vertx.mutiny.pgclient.PgPool;

public interface PratoService {
    public Multi<PratoDTO> buscarPratos(PgPool pgPool);
    public Multi<PratoDTO> buscarPratoPorId(PgPool pgPool, Long id);
    public Multi<PratoDTO> buscarPratosPorIdRestaurante(PgPool pgPool, Long restauranteId);
}