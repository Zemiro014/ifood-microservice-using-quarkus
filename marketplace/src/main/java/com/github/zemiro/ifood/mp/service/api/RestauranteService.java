package com.github.zemiro.ifood.mp.service.api;

import com.github.zemiro.ifood.mp.referenceEntity.Restaurante;

public interface RestauranteService {

    public void salvarRestaurante(Restaurante entity);
}