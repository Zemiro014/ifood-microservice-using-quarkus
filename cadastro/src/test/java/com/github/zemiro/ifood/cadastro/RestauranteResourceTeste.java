package com.github.zemiro.ifood.cadastro;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import com.github.zemiro.ifood.cadastro.utils.CadastroTestLifecycleManager;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;
import static org.hamcrest.CoreMatchers.is;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTeste {

    @Test
    @DataSet("restaurantes-ceneario-1.yml")
    public void testeBuscarrestaurante() {
        String resultado = given()
          .when().get("/restaurantes")
          .then()
             .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(resultado);
    }

}