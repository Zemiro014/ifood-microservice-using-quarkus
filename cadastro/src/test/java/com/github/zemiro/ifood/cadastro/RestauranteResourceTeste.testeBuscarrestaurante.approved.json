[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "proprietario": "ID do KleyCloack"
  },
  {
    "id": 124,
    "nome": "Novo Restaurante 2",
    "proprietario": "ID do KleyCloack 2"
  }
]